packages/block-editor/src/components/link-control/index.js

```diff
@@ -100,6 +103,33 @@ function LinkControl( {
 	}, [] );
 	const displayURL = ( value && filterURLForDisplay( safeDecodeURI( value.url ) ) ) || '';
 
+	useEffect( () => {
+		// When `isEditingLink` is set to `false`, a focus loss could occur
+		// since the link input may be removed from the DOM. To avoid this,
+		// reinstate focus to a suitable target if focus has in-fact been lost.
+		const hadFocusLoss = (
+			isEndingEditWithFocus.current &&
+			wrapperNode.current &&
+			! wrapperNode.current.contains( document.activeElement )
```

Comment:
Do we really need to check for the position of the focus? or is this something that should be done every time the edit mode is closed?

Reply:
> Do we really need to check for the position of the focus? or is this something that should be done every time the edit mode is closed?

There's a bit more explanation in the code comment below:

>Note that typically this would be the read-only mode's link element, but isn't guaranteed to be, since the input may continue to be forced to be shown if the next value is still unassigned.

After the editing state is toggled, there might not actually be a focus loss, depending if the next render would have a usable value. If it doesn't have a usable value, the link input would continue to be shown.

I guess the comment doesn't make sense as much sense in the context of the focus loss condition.

Reply:
Shifted around and reworded the comment in 0d6bf78.